src/back/controllers/actions/user.js

```diff
@@ -21,7 +21,29 @@ function watchChallenge(userPromise, challengeId) {
     return userPromise.then(user => {
         user.set(
             'watchList',
-            user.get('watchList').reduce((list, id) => [...list, id], [challengeId])
+            user
+                .get('watchList')
+                .filter(taskId => taskId.toString() !== challengeId)
+                .reduce((list, id) => [...list, id], [challengeId])
+        );
+        return user.save();
+    });
+}
+
+function acceptChallenge(userPromise, challengeId) {
```

Comment:
@wips I mean, that userPromise is unnecessary argument. You can get something like 
```javascript 
function acceptChallenge(challengeId) {
    const STATUS_ACCEPTED = 'accepted';
    const newlyAccepted = {
        status: STATUS_ACCEPTED,
        date: new Date(),
        challenge: challengeId
    };
    return (user) => {
        user.set(
            'challenges',
            user
                .get('challenges')
                .filter(task => task.status !== STATUS_ACCEPTED && challengeId !== task.challenge)
                .reduce((list, task) => [...list, task], [newlyAccepted])
        );
        return user.save();
    };
}
```
Also, in that case you can use $push to add challenge and pull to remove. In my opinion it is better then get all challenges from user to server.